File: src/jamtaba/ninjam/protocol/server/messages/UserChannel.java
package jninjam.protocol.server.messages;

import jninjam.NinjaMUser;

/**
 *
 * @author zeh
 */
public class UserChannel {

    private final NinjaMUser user;
    private String name;
    private boolean active;
    private final int index;
    private short volume;//(dB gain, 0=0dB, 10=1dB, -30=-3dB, etc)
    private byte pan;//Pan [-128, 127]
    private byte flags;

    @Override
    public String toString() {
        return "UserChannel{" + "name=" + name + ", active=" + active + ", index=" + index + ", volume=" + volume + ", pan=" + pan + ", flags=" + getFlags() + '}';
    }

    public UserChannel(NinjaMUser user, String name, boolean active, int channelIndex, short volume, byte pan, byte flags) {
        this.user = user;
        this.name = name;
        this.active = active;
        this.index = channelIndex;
        this.volume = volume;
        this.pan = pan;
        this.flags = flags;
    }

    @Override
    public int hashCode() {
        int hash = 3;
        hash = 37 * hash + (this.user != null ? this.user.hashCode() : 0);
        hash = 37 * hash + this.index;
        return hash;
    }

    public boolean isActive() {
        return active && flags == 0;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final UserChannel other = (UserChannel) obj;
        if (this.user != other.user && (this.user == null || !this.user.equals(other.user))) {
            return false;
        }
        if (this.index != other.index) {
            return false;
        }
        return true;
    }

    

    public int getIndex() {
        return index;
    }
    
    public String getName() {
        return name;
    }

    public NinjaMUser getUser() {
        return user;
    }

    public void setName(String name) {
        this.name = name;
    }

    public byte getFlags() {
        return flags;
    }

    public void setFlags(byte flags) {
        this.flags = flags;
    }

}

File: src/ninjam/UserChannel.cpp
#include "UserChannel.h"

#include <QDebug>

using namespace Ninjam;

UserChannel::UserChannel(QString userFullName, QString channelName, bool active, int channelIndex,
                         short volume, quint8 pan, quint8 flags) :
    userFullName(userFullName),
    channelName(channelName),
    active(active),
    index(channelIndex),
    volume(volume),
    pan(pan),
    flags(flags)
{
}

UserChannel::UserChannel() :
    userFullName("invalid"),
    channelName("invalid"),
    active(false),
    index(-1),
    volume(0),
    pan(0),
    flags(0)
{
}

UserChannel::UserChannel(const UserChannel &c) :
    userFullName(c.getUserFullName()),
    channelName(c.getName()),
    active(c.isActive()),
    index(c.getIndex()),
    volume(c.volume),
    pan(c.pan),
    flags(c.flags)
{
}

UserChannel::~UserChannel()
{
}
